src/Components/Web.JS/src/Boot.Server.ts

```diff
@@ -37,8 +37,13 @@ async function boot(userOptions?: Partial<BlazorOptions>): Promise<void> {
     }
 
     const reconnection = existingConnection || await initializeConnection(options, logger);
+    if (reconnection.state !== signalR.HubConnectionState.Connected) {
+      logger.log(LogLevel.Information, 'Reconnection attempt failed. Unable to connect to the server.');
+      return false;
+    }
+
     if (!(await circuit.reconnect(reconnection))) {
```

Comment:
SignalR throws if you attempt to reconnect like this while the connection is not connected. This prevented the reconnect display from properly handling scenarios where the server went away.

Reply:
Actually, can't do this. It breaks the expectation of calling code, i.e. in `DefaultReconnectionHandler` it expects:

```
        // reconnectCallback will asynchronously return:
        // - true to mean success
        // - false to mean we reached the server, but it rejected the connection (e.g., unknown circuit ID)
        // - exception to mean we didn't reach the server (this can be sync or async)
```

In a way it makes sense. Basically, we need to have 3 different return values and instead of having an enum the previous design utilized exceptions. I'll revert this change in my next change for this experience.